doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+    target_pose.position.y = -0.49;
+    target_pose.position.z = 0.38;
+    move_group.setPoseTarget(target_pose);
+
+    // Rviz communication
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Constrained Goal", rvt::WHITE, rvt::XLARGE);
+    visual_tools.publishAxisLabeled(target_pose, "point");
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' to plan a constrained path");
+
+    // BEGIN_SUB_TUTORIAL constrained_path
+    // Planning with Path Constraints
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // Path constraints can easily be specified for a link on the robot.
+    // Let's specify a path constraint for our group.
+    moveit_msgs::OrientationConstraint ocm;
+    ocm.link_name = "panda_link7";
+    ocm.header.frame_id = "panda_link0";
+    ocm.orientation.w = 0.0;
+    ocm.orientation.x = 0.383;
+    ocm.orientation.y = -0.924;
+    ocm.orientation.z = -0.0;
+    ocm.absolute_x_axis_tolerance = 0.1;
+    ocm.absolute_y_axis_tolerance = 0.1;
+    ocm.absolute_z_axis_tolerance = 0.1;
+    ocm.weight = 1.0;
+
+    // Now, set it as the path constraint for the group.
+    moveit_msgs::Constraints test_constraints;
+    test_constraints.orientation_constraints.push_back(ocm);
+    move_group.setPathConstraints(test_constraints);
+
+    // Planning with constraints can be slow because every sample must call an inverse kinematics
+    // solver. Lets increase the planning time from the default 5 seconds to be sure the planner
+    // has enough time to succeed.
+    move_group.setPlanningTime(10.0);
+
+    // Now we will plan to a new pose. As a note both the start and the end of the path should
+    // follow the constraint rules
+    move_group.setPoseTarget(target_pose);
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    // When done with the path constraint be sure to clear it.
+    move_group.clearPathConstraints();
+
+    // For comparison an unconstrained path to this point can be made by planning to the already set Pose target.
+    //
+    // END_SUB_TUTORIAL
+
+    // RViz
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 3 (constraints) %s", success ? "" : "FAILED");
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' to compare the constrained path with an unconstrained path");
+  }
+
+  void plan_unconstrained()
+  {
+    // This plans to the same existing goal
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 4 (comparison) %s", success ? "" : "FAILED");
+
+    // RViz, This time we don't delete the prevous marks to shows the two paths side by side
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.trigger();
+  }
+
+  void plan_cartesian_path()
```

Comment:
This should take in a vector of target_goals and return the fraction of success. Execution and visualization should be done in a separate class that get's re-used. Make things modular and treat this more like a high level class for interacting with move_group.